sdks/java/core/src/main/java/org/apache/beam/sdk/io/AvroWriteSchemaTransformProvider.java

```diff
@@ -0,0 +1,141 @@
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.io;
+
+import com.google.auto.value.AutoValue;
+import java.io.Serializable;
+import java.util.Collections;
+import java.util.List;
+import javax.annotation.Nullable;
+import org.apache.avro.generic.GenericRecord;
+import org.apache.beam.sdk.schemas.AutoValueSchema;
+import org.apache.beam.sdk.schemas.Schema;
+import org.apache.beam.sdk.schemas.annotations.DefaultSchema;
+import org.apache.beam.sdk.schemas.transforms.Convert;
+import org.apache.beam.sdk.schemas.transforms.SchemaTransform;
+import org.apache.beam.sdk.schemas.transforms.TypedSchemaTransformProvider;
+import org.apache.beam.sdk.schemas.utils.AvroUtils;
+import org.apache.beam.sdk.transforms.PTransform;
+import org.apache.beam.sdk.transforms.windowing.FixedWindows;
+import org.apache.beam.sdk.transforms.windowing.Window;
+import org.apache.beam.sdk.values.PCollection;
+import org.apache.beam.sdk.values.PCollectionRowTuple;
+import org.apache.beam.vendor.guava.v26_0_jre.com.google.common.base.Strings;
+import org.joda.time.Duration;
+
+public class AvroWriteSchemaTransformProvider
+    extends TypedSchemaTransformProvider<
+        AvroWriteSchemaTransformProvider.AvroWriteSchemaTransformConfiguration> {
+  @Override
+  public String identifier() {
+    return "beam:schematransform:org.apache.beam:avro_write:v1";
+  }
+
+  @Override
+  public List<String> inputCollectionNames() {
+    return Collections.singletonList("input");
+  }
+
+  @Override
+  public List<String> outputCollectionNames() {
+    return Collections.emptyList();
+  }
+
+  @Override
+  protected Class<AvroWriteSchemaTransformConfiguration> configurationClass() {
+    return AvroWriteSchemaTransformConfiguration.class;
+  }
+
+  @Override
+  protected SchemaTransform from(AvroWriteSchemaTransformConfiguration configuration) {
+    return new AvroWriteSchemaTransform(configuration);
+  }
+
+  static class AvroWriteSchemaTransform implements SchemaTransform, Serializable {
+
+    AvroWriteSchemaTransformConfiguration config;
+
+    AvroWriteSchemaTransform(AvroWriteSchemaTransformConfiguration config) {
+      this.config = config;
+    }
+
+    @Override
+    public PTransform<PCollectionRowTuple, PCollectionRowTuple> buildTransform() {
```

Comment:
Same concerns as for this part in the ReadSTProvider.

Reply:
Same